File: hot-framework/src/test/resources/be/solidx/hot/promise/promises2.js

var promise = deferred.promise()

promise.then(function(result) {
	return subdeferred.promise();
}).then(function(result) {
	return result * 3
}).done(function(result) {
	hprint(result)
})

deferred.resolve("hello")
subdeferred.resolve(30)

File: hot-framework/src/test/resources/be/solidx/hot/promise/promises2.py
def subThen(result):
    return result*3

def done(result):
    print result

def then(result):
    return subdeferred.promise()

def fail(e):
    print e

def thenfail(message):
    return message + message

def always():
    print "always"

promise = deferred.promise()
promise.then(then).then(subThen).done(done).always(always)

subdeferred.resolve(20)
deferred.resolve(10)
